### ISSUE TITLE
incorrect uv map after mesh smoothed

### ISSUE THREAD

**Issue by cosw0t** [mikedh/trimesh#1403]
I'm using the excellent implementation of `Trimesh.smoothed` to remove some ugly artefacts in my assets, however it seems that the uv map is either lost (error in concatenating visuals if images are None) or gives the wrong UVs.

- w.r.t. the error in concatenating visuals, which occurs here https://github.com/mikedh/trimesh/blob/main/trimesh/path/packing.py#L429
where None image gives no size, therefore it fails (and continues without Texture materials). Is it possible to ignore the fact that some images are None? the uvs should could be still concatenated.

- w.r.t. to the wrong UVS in the end for the example below, i have a mesh with 2098 vertices and 191015 uvs 
my 2 cents go on this line https://github.com/mikedh/trimesh/blob/main/trimesh/util.py#L1590, which could instead select the right uvs. However i can't see an obvious way in the short time i've looked at this

any hints?

code below:

```
import numpy as np
import trimesh
from PIL import Image

mesh = trimesh.load("./teapot.obj").unwrap(Image.fromarray(np.empty([512, 512, 3], dtype=np.uint8)))

mesh = mesh.smoothed()
print("vertices:", len(mesh.vertices))
print("uvs:", len(mesh.visual.uv))
```

> vertices: 2098
> uvs: 191015

[teapot.zip](https://github.com/mikedh/trimesh/files/7468978/teapot.zip)

**Comment by michele-arrival** [comment]
Actually replacing
```
    visuals.append(mesh.visual.face_subset(index))
```

with 

```
        vis = mesh.visual.face_subset(index)
        vis.update_vertices(unique)
        visuals.append(vis)
```

seem to do the trick in my specific case. Is there anything obviously wrong with this?

**Comment by mikedh** [comment]
Oh interesting, thanks for the PR! I suspect this and #1395 are historical (texture was implemented much later than `subset`, `split`, etc). 

Fix looks good, my only question is is whether that logic should be moved upwards to `visual.face_subset`?

**Comment by michele-arrival** [comment]
yes, that's the first place where i looked but couldn't find an obvious solution.

either there should be a new function like `vertex_subset`. However `update_vertices` does something similar (it just doesn't return anything, hence the 2 steps in my fix)

or maybe the `face_subset` should compute vertex indices from the `self.mesh.faces` and fix the vertex subset too in all classes? 
it would fix the uvs in the `TextureVisuals` and the `vertex_colors` in `ColorVisulas`, which i don't see this handled in any of 
the Visuals classes. same for `VertexColor` class.

**Comment by michele-arrival** [comment]
maybe something like this? https://github.com/mikedh/trimesh/pull/1405/files
